gflownet/proxy/crystals/composition.py

```diff
@@ -0,0 +1,73 @@
+import pickle
+from pathlib import Path
+
+import numpy as np
+import torch
+from torchtyping import TensorType
+
+from gflownet.proxy.base import Proxy
+
+PROTONS_NUMBER_COUNTS = None
+
+
+def _read_protons_number_counts():
+    global PROTONS_NUMBER_COUNTS
+    if PROTONS_NUMBER_COUNTS is None:
+        with open(
+            Path(__file__).parent / "number_of_protons_counts.pkl", "rb"
+        ) as handle:
+            PROTONS_NUMBER_COUNTS = pickle.load(handle)
+    return PROTONS_NUMBER_COUNTS
+
+
+class CompositionMPFrequency(Proxy):
+    def __init__(self, normalise: bool = True, **kwargs):
+        super().__init__(**kwargs)
+        self.counts_dict = _read_protons_number_counts()
+        self.normalise = normalise
+
+    def _get_max_protons_state(self, env):
+        max_protons_state = env.source.copy()
+        for elem in env.required_elements:
+            max_protons_state[env.elem2idx[elem]] = max(1, env.min_atom_i)
+        for idx in range(env.min_diff_elem - len(env.required_elements)):
```

Comment:
What if given element was already set to something other than zero before (or in other words, last `env.min_diff_elem - len(env.required_elements)` elements included some of the `required_elements`? Wouldn't this try to set them twice, leading to not enough different elements being used?

Reply:
@AlexandraVolokhova ?

Reply:
Good catch! I've fixed that and added tests where this occurs